### DISCUSSION TITLE
Filter .or with and() not working

### DISCUSSION THREAD

**Discussion by mmmurilo** [supabase/supabase#12779] [Questions]
Hi, i'm trying to query my view with this query:
query = query.or(`seller_ids.cs.{${filter.address_origin_seller}},and(organisation_address_id.is.null)`)

When i insert the and condition the is ignoring what i filtered before and don't respect the seller_ids.cs condition.
Any ideia of what is wrong?

**Comment by steve-chavez** [comment]
Shouldn't it be like this?

```js
query.or(seller_ids.cs.{${filter.address_origin_seller}},organisation_address_id.is.null)
```

**Reply by mmmurilo** [reply]
I need both conditions combined, not only one. My complete query i want actualy that the filter looks for seller_ids only if organisation_address_id is null, if isn't null i want compare by the organisation_address_id. I have tried this:
query.contains('seller_ids', filter.address_origin_seller).
is('organisation_address_id', null)
.or(`organisation_address_id.eq.${filter.address_origin_id}`)

But the .or condition is never satisfied.

**Reply by mmmurilo** [reply]
This is the WHERE condition i want select:
```
WHERE (orders.seller_ids @> my_seller_id AND reverse_trackings.organisation_address_id IS NULL)
   OR (reverse_trackings.organisation_address_id = my_address_id)
```

**Reply by steve-chavez** [reply]
I see. That one should be:

```js
query.or(`and(seller_ids.cs.{${filter.address_origin_seller}},organisation_address_id.is.null), reverse_trackings.organisation_address_id.eq.my_address_id`)
```
